torch/utils/benchmark/utils/timer.py

```diff
@@ -13,27 +14,109 @@
+
+    4) Instruction counts
+        In addition to wall times, Timer can run a statement under Callgrind
+        and report instructions executed.
+
+    Arguments:
+        Directly analogous to `timeit.Timer` constructor arguments:
+        -----------------------------------------------------------------------
+        stmt: Code snippet to be run in a loop and timed.
+        setup: Optional setup code. Used to define variables used in `stmt`
+        timer:
+            Callable which returns the current time. If PyTorch was built
+            without CUDA or there is no GPU present, this defaults to
+            `timer.default_timer`; otherwise it will synchronize CUDA before
+            measuring the time.
+
+        globals:
+            A dict which defines the global variables when `stmt` is being
+            executed. This is the other method for providing variables which
+            `stmt` needs.
+
+        PyTorch Timer specific constructor arguments:
+        -----------------------------------------------------------------------
+        label:
+            String which summarizes `stmt`. For instance, if `stmt` is
+              "torch.nn.functional.relu(torch.add(x, 1, out=out))"
+            one might set label to "ReLU(x + 1)" to improve readability.
+
+        sub_label:
+            Provide supplemental information to disambiguate measurements
+            with identical stmt or label. For instance, in our example
+            above sub_label might be "float" or "int", so that it is easy
+            to differentiate:
+                "ReLU(x + 1): (float)"
+                "ReLU(x + 1): (int)"
+            when printing Measurements or summarizing using `Compare`.
+
+        description:
+            This is a tag which is completely orthoganal to stmt, label, and
+            sub_label. (For instance a description of the particular size or
+            layout of the data that `stmt` is operating on.) The principle
+            use of `description` is to signal to `Compare` the columns of
+            data. It also appears in the __repr__ of `Measurement`
+
+        env:
+            This tag indicates that otherwise identical tasks were run in
+            different environments, and are therefore not equivilent. For
+            instance, when A/B testing a change to a kernel. `Compare` will
+            treat Measurements with different `env` specification as distinct
+            when merging replicate runs.
+
+        num_threads:
+            The size of the PyTorch threadpool when executing `stmt`. Single
+            threaded performace is important as both a key inference workload
+            and a good indicator of intrinsic algorithmic efficiency, so the
+            default is set to one. This is in contrast to the default PyTorch
+            threadpool size which tries to utilize all cores so that most
+            users don't have to think about threading in most cases.
```

Comment:
heh, given the number of issues when users do have do think about threading I'd drop this sentence.

Reply:
Heh.